atomic_reactor/build.py

```diff
@@ -204,6 +204,11 @@ def set_df_path(self, path):
                 custom_base_images.add(image_str)
                 self.custom_parent_image = True
             self.parents_ordered.append(image_str)
+
+            if self.original_base_image.to_str() == image:
```

Comment:
Shouldn't we compare with `image_str` instead of `image`

Also this part of code is pretty obfuscated to me, there are no comments and I don't know why we are setting `None`

Reply:
Is `self.original_base_image` different from `image_name` in this case? If not, isn't this already being set to `None` below?

Reply:
> Is `self.original_base_image` different from `image_name` in this case? If not, isn't this already being set to `None` below?

The problem is that they are two different *instances*, and so they have different keys.

Reply:
Oh, `image` is a string, I get it now.

Reply:
@MartinBasti so, `None` here means that there is a parent image we know about, and we have no local copy of it yet.

Reply:
/me would give a treasure for that comment in code